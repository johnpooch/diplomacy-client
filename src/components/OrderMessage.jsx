import React from 'react';

function shouldRender(order) {
  const { type, target } = order;
  const possibleOrderTypes = ['move', 'retreat', 'support', 'convoy'];
  return target ? false : possibleOrderTypes.includes(type);
}

function getPreposition(order) {
  const { type, aux } = order;
  if (['move', 'retreat'].includes(type)) {
    return 'into';
  }
  return aux ? 'into' : 'from';
}

const OrderMessage = (props) => {
  const { order } = props;
  const { type } = order;
  if (!shouldRender(order)) return null;
  const preposition = getPreposition(order);
  return (
    <div>
      Select a territory to {type} {preposition}.
    </div>
  );
};

export default OrderMessage;
