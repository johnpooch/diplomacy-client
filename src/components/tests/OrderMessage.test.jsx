import React from 'react';
import { render } from '@testing-library/react';

import OrderMessage from '../OrderMessage';

const aux = {
  piece: {
    type: 'army',
  },
  territory: {
    name: 'york',
  },
};

const target = {
  territory: {
    name: 'wales',
  },
};

test('Build', () => {
  const order = {
    type: 'build',
    aux: null,
    target: null,
  };
  const { asFragment } = render(<OrderMessage order={order} />);
  expect(asFragment()).toMatchSnapshot();
});

test('Hold', () => {
  const order = {
    type: 'hold',
    aux: null,
    target: null,
  };
  const { asFragment } = render(<OrderMessage order={order} />);
  expect(asFragment()).toMatchSnapshot();
});

test('Disband', () => {
  const order = {
    type: 'disband',
    aux: null,
    target: null,
  };
  const { asFragment } = render(<OrderMessage order={order} />);
  expect(asFragment()).toMatchSnapshot();
});

test('Move without target', () => {
  const order = {
    type: 'move',
    aux: null,
    target: null,
  };
  const { asFragment } = render(<OrderMessage order={order} />);
  expect(asFragment()).toMatchSnapshot();
});

test('Move with target', () => {
  const order = {
    type: 'move',
    aux: null,
    target,
  };
  const { asFragment } = render(<OrderMessage order={order} />);
  expect(asFragment()).toMatchSnapshot();
});

test('Support without aux', () => {
  const order = {
    type: 'support',
    aux: null,
    target: null,
  };
  const { asFragment } = render(<OrderMessage order={order} />);
  expect(asFragment()).toMatchSnapshot();
});

test('Support with aux', () => {
  const order = {
    type: 'support',
    aux,
    target: null,
  };
  const { asFragment } = render(<OrderMessage order={order} />);
  expect(asFragment()).toMatchSnapshot();
});

test('Support with target', () => {
  const order = {
    type: 'support',
    aux,
    target,
  };
  const { asFragment } = render(<OrderMessage order={order} />);
  expect(asFragment()).toMatchSnapshot();
});
