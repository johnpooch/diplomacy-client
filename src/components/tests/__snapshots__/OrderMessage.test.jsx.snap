// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Build 1`] = `<DocumentFragment />`;

exports[`Disband 1`] = `<DocumentFragment />`;

exports[`Hold 1`] = `<DocumentFragment />`;

exports[`Move with target 1`] = `<DocumentFragment />`;

exports[`Move without target 1`] = `
<DocumentFragment>
  <div>
    Select a territory to move into.
  </div>
</DocumentFragment>
`;

exports[`Support with aux 1`] = `
<DocumentFragment>
  <div>
    Select a territory to support into.
  </div>
</DocumentFragment>
`;

exports[`Support with target 1`] = `<DocumentFragment />`;

exports[`Support without aux 1`] = `
<DocumentFragment>
  <div>
    Select a territory to support from.
  </div>
</DocumentFragment>
`;
